bin/pycbc_compress_bank

```diff
@@ -0,0 +1,172 @@
+                     "waveforms for checking precision. Options are 'linear', "
+                     "or any interpolation recognized by scipy's interp1d "
+                     "kind argument. Default is linear.")
+parser.add_argument("--verbose", action="store_true", default=False)
+
+
+args = parser.parse_args()
+
+pycbc.init_logging(args.verbose)
+
+fmin = args.low_frequency_cutoff
+fmax = args.sample_rate/2.
+
+# load the bank
+logging.info("loading bank")
+# we'll just use dummy values for N, df for now
+bank = waveform.FilterBank(args.bank_file, 5, 0.25, fmin, numpy.complex128,
+    approximant=args.approximant)
+templates = bank.table
+if args.tmplt_index is not None:
+    imin, imax = args.tmplt_index
+    templates = templates[imin:imax]
+else:
+    imin, imax = 0, templates.size
+
+# figure out the dfs needed for each waveform
+logging.info("getting needed dfs")
+# we'll ensure that the there are atleast 2 samples in a waveform
+seg_lens = 4*numpy.array([max(4./args.sample_rate,
+    2**numpy.ceil(numpy.log2(compress.rough_time_estimate(m1, m2, fmin))))
+    for m1,m2 in zip(templates.mass1, templates.mass2)])
+
+# generate output file
+# FIXME: this should be moved to a class or function
+logging.info("writing template info to output")
+output = h5py.File(args.output, 'w')
+output['mass1'] = templates['mass1']
+output['mass2'] = templates['mass2']
+output['spin1z'] = templates['spin1z']
+output['spin2z'] = templates['spin2z']
+output['template_hash'] = templates['template_hash']
+
+# get the compressed sample points for each template
+logging.info("getting compressed amplitude and phase")
+mismatches = numpy.zeros(imax-imin, dtype=float)
+for ii in range(imin, imax):
+    # update the N, df to use based on this waveform
+    N = int(args.sample_rate*seg_lens[ii-imin])
+    df = 1./seg_lens[ii-imin]
+    bank.delta_f = df
+    bank.N = N
+    bank.filter_length = N/2 + 1
+    # We'll budge the waveform starting frequency down by 1df to ensure 
+    # the first point is correct
+    bank.f_lower = fmin-df
+    bank.kmin = int(bank.f_lower / df)
+    # scratch space
+    decomp_scratch = FrequencySeries(numpy.zeros(N, dtype=numpy.complex128),
+                                 delta_f=df)
+    # generate the waveform
+    htilde = bank[ii]
+    tmplt = bank.table[ii]
+    kmax = htilde.end_idx
+    
+    # get the compressed sample points
+    if args.compression_algorithm == 'mchirp':
+        sample_points = compress.mchirp_compression(tmplt.mass1, tmplt.mass2,
+            fmin, (kmax-1)*df, min_seglen=args.t_pad, df_multiple=df)
+    elif args.compression_algorithm == 'spa':
+        sample_points = compress.spa_compression(htilde, fmin, (kmax-1)*df,
+            min_seglen=args.t_pad)
+    else:
+        raise ValueError("unrecognized compression algorithm %s" %(
+            args.compression_algorithm))
+            
+    # compress
+    sample_points, comp_amp, comp_phase, mismatch = compress.compress_waveform(
```

Comment:
Is there a compress.uncompress_waveform? This would take the compressed waveform and the original sample rate and produce (almost) the original waveform. In fact a good sanity check would be to run compress.compress_waveform and then compress.uncompress_waveform and verify that the uncompressed waveform and the original waveform have overlap within the tolerance. Can this be added to the test/ directory before pushing if such sanity checks are not already there?